src/Microsoft.AspNet.SignalR.Redis/RedisScaleoutOptions.cs

```diff
@@ -19,18 +20,8 @@ public RedisScaleoutOptions(string server, int port, string password, string eve
         {
         }
 
-        public RedisScaleoutOptions(string connectionString, string eventKey)
+        public RedisScaleoutOptions([NotNull] string connectionString, [Notnull] string eventKey)
         {
-            if (String.IsNullOrEmpty(connectionString))
-            {
-                throw new ArgumentNullException("connectionString");
-            }
-
-            if (String.IsNullOrEmpty(eventKey))
-            {
-                throw new ArgumentNullException("eventKey");
-            }
-
```

Comment:
- This code should instead `throw` an `ArgumentException` with something like the fairly-common `Resources.ArgumentCannotBeNullOrEmpty`.  We do _not_ use `[NotNull]` in this case so that the `null` and `string.Empty` cases leads to the same `Exception` type and `Message`.
- Of course `nameof()` would be great in the updated code.